### ISSUE TITLE
Feature: Preact support

### ISSUE THREAD

**Issue by kaicataldo** [kaicataldo/eleventy-plugin-react#7]
In the spirit of keeping everything as light and fast as possible, we should allow for native support of Preact!

**Comment by kaicataldo** [comment]
Yeah, it could be worth expanding the scope of this plugin to `eleventy-plugin-jsx` at some point...

**Comment by kaicataldo** [comment]
@SimeonGriggs FWIW, I also published https://github.com/kaicataldo/eleventy-plugin-jsx. This is only for using JSX to generate static HTML, though, so not sure if the addition of Preact in it is a huge win.

**Comment by SimeonGriggs** [comment]
That's cool, love it.

**Comment by kaicataldo** [comment]
With a big focus of this project being to balance DX with performance, I do think moving to Preact is inline with the project goals. Might be worth exploring sooner rather than later! If we do want to do that, does it make sense to combine this and `eleventy-plugin-jsx` into one project? I think we'd probably drop support for React and make this a more batteries-included experience.
